consensus/hotstuff/voteaggregator/vote_aggregator_v2.go

```diff
@@ -185,8 +185,25 @@ func (va *VoteAggregatorV2) AddBlock(block *model.Proposal) error {
 	return nil
 }
 
-func (va *VoteAggregatorV2) InvalidBlock(block *model.Proposal) error {
-	panic("implement me")
+// InvalidBlock notifies the VoteAggregator about an invalid proposal, so that it can process votes for the invalid
+// block and slash the voters.
+func (va *VoteAggregatorV2) InvalidBlock(proposal *model.Proposal) error {
+	slashingVoteConsumer := func(vote *model.Vote) {
+		if proposal.Block.BlockID == vote.BlockID {
+			va.notifier.OnVoteForInvalidBlockDetected(vote, proposal)
+		}
+	}
+
+	block := proposal.Block
+	collector, _, err := va.collectors.GetOrCreateCollector(block.View)
+	if err != nil {
+		return fmt.Errorf("could not retrieve collector for invalid proposal (%x) at view %d: %w",
+			block.BlockID, block.View, err)
```

Comment:
`VoteCollector` collects votes for a specific view (not a block)

```suggestion
		return fmt.Errorf("could not retrieve vote collector for view %d: %w", block.View, err)
```